minicluster/src/main/java/alluxio/underfs/UnderFileSystemCluster.java

```diff
@@ -71,7 +71,7 @@ public static synchronized UnderFileSystemCluster get() {
    */
   public static synchronized UnderFileSystemCluster get(String baseDir, Configuration configuration)
       throws IOException {
-    if (sUnderFSCluster == null) {
+    if (sUnderFSCluster == null || sUnderFSCluster.mBaseDir != baseDir) {
```

Comment:
This is a separate bug where we would use the same UFS cluster for every test, even though the base directory would change between tests. The UFS cluster would use the base directory created/deleted by the first unit test it a suite.